docs/PLUGIN_DIAGNOSTICS.md

```diff
@@ -0,0 +1,114 @@
+# Plugin Diagnostics
+
+Plugin diagnostics provides a simple way to verify that a plugin is capable of running without requiring the plugin to be loaded and a task started.
+
+It is possible to run plugin diagnostics without any arguments, e.g.:
+
+```
+$ ./snap-plugin-collector-psutil
+```
+
+or with plugin configuration in the form of a JSON:
+
+```
+$ ./snap-plugin-collector-psutil --config {\"mouting_points\":\"/dev/sda\"}
+```
+
+Plugin diagnostics delivers following information:
+- Runtime details, e.g.:
```

Comment:
@katarzyna-z, what do you think about listing all information one by one, and below provide just exemplary output from plugin diagnostics?

Plugin diagnostics delivers following information:
- Runtime details (plugin name, version, etc.)
- Configuration details 
- List of metrics expose by the plugin
- Metrics with their values that can be collected right now